File: trunk/mst-platform/src/java/xc/mst/action/user/EditMyAccount.java
/**
  * Copyright (c) 2009 eXtensible Catalog Organization
  *
  * This program is free software; you can redistribute it and/or modify it under the terms of the MIT/X11 license. The text of the
  * license can be found at http://www.opensource.org/licenses/mit-license.php and copy of the license can be found on the project
  * website http://www.extensiblecatalog.org/.
  *
  */

package xc.mst.action.user;

import org.apache.log4j.Logger;

import xc.mst.action.BaseActionSupport;
import xc.mst.action.UserAware;
import xc.mst.bo.user.User;
import xc.mst.constants.Constants;
import xc.mst.dao.DataException;

/**
 * Action to edit user account
 *
 * @author Sharmila Ranganathan
 *
 */
public class EditMyAccount extends BaseActionSupport implements UserAware {

	/** Generated id  */
	private static final long serialVersionUID = 1117303971697447644L;

	/** A reference to the logger for this class */
	static Logger log = Logger.getLogger(Constants.LOGGER_GENERAL);

	/** New user registering with the system */
	private User user;

	/** First name of user */
	private String firstName;

	/** Last name of user */
	private String lastName;
	
	/** Email of user */
	private String email;

	/** Error type */
	private String errorType; 
	
	/** Information message to user */
	private String message;
	
	/**
	 * To view edit my account
	 */
    @Override
	public String execute() throws DataException {
		firstName =  user.getFirstName();
		lastName =  user.getLastName();
		email = user.getEmail();

		return SUCCESS;
	}

	/**
	 * Save user account information
	 *
	 * @return
	 */
	public String saveMyAccount() {

		log.debug(EditMyAccount.class + ":" + "saveMyAccount()" );

		try {
				User userWithEmail = getUserService().getUserByEmail(email, user.getServer());
				if (userWithEmail == null || userWithEmail.getId() == user.getId()) {
					user.setFirstName(firstName);
					user.setLastName(lastName);
					user.setEmail(email);

					getUserService().updateUser(user);
					errorType = "info";
					message = "Account information saved.";
				} else {
					addFieldError("emailExist", "Email already exists - " + email);
					errorType = "error";
					return INPUT;
				}
		} catch (Exception e) {
			log.error("Exception occured while saving user account information", e);
			addFieldError("dataError", e.getMessage());
			errorType = "error";
			return INPUT;
		}
		return SUCCESS;
	}



	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getErrorType() {
		return errorType;
	}

	public void setErrorType(String errorType) {
		this.errorType = errorType;
	}

	public String getFirstName() {
		return firstName;
	}

	public void setFirstName(String firstName) {
		this.firstName = firstName.trim();
	}

	public String getLastName() {
		return lastName;
	}

	public void setLastName(String lastName) {
		this.lastName = lastName.trim();
	}

	public String getMessage() {
		return message;
	}


}

File: mst-platform/src/webapp/st/page-resources/js/edit_my_account.js
 /*
  * Copyright (c) 2009 eXtensible Catalog Organization
  *
  * This program is free software; you can redistribute it and/or modify it under the terms of the MIT/X11 license. The text of the
  * license can be found at http://www.opensource.org/licenses/mit-license.php and copy of the license can be found on the project
  * website http://www.extensiblecatalog.org/.
  *
  */

YAHOO.namespace("xc.mst.account");

YAHOO.xc.mst.account = {

  save : function()
  {

    if (document.getElementById('user_first_name').value=='') {
      createErrorDiv("error",'User first Name is required.');
      return false;
    }

    if (document.getElementById('user_last_name').value=='') {
      createErrorDiv("error",'User last Name is required.');
      return false;
    }
    if (document.getElementById('user_email').value=='') {
      createErrorDiv("error",'Email is required.');
      return false;
    } else {
      if (!YAHOO.xc.mst.account.emailCheck(document.getElementById('user_email').value)) {
        createErrorDiv("error","Invalid E-mail ID");
        return false;
      }

    }

    document.myAccountForm.action=  'saveMyAccount.action';
    document.myAccountForm.submit();

  },

  cancel : function()
  {
    document.myAccountForm.action=  'allRepository.action';
    document.myAccountForm.submit();

  },

  changePassword : function(userId)
  {
    document.myAccountForm.action=  'viewChangePassword.action';
    document.myAccountForm.submit();

  },

  emailCheck : function(str)
  {
      var at="@"
      var dot="."
      var lat=str.indexOf(at)
      var lstr=str.length
      var ldot=str.indexOf(dot)
      if (str.indexOf(at)==-1){
         return false
      }

      if (str.indexOf(at)==-1 || str.indexOf(at)==0 || str.indexOf(at)==lstr){
         return false
      }

      if (str.indexOf(dot)==-1 || str.indexOf(dot)==0 || str.indexOf(dot)==lstr){
          return false
      }

       if (str.indexOf(at,(lat+1))!=-1){
          return false
       }

       if (str.substring(lat-1,lat)==dot || str.substring(lat+1,lat+2)==dot){
          return false
       }

       if (str.indexOf(dot,(lat+2))==-1){
          return false
       }

       if (str.indexOf(" ")!=-1){
          return false
       }

        return true
  }


}

